content/courses/advanced-ada/chapters/access_types.rst

```diff
@@ -294,14 +294,289 @@ Unchecked Access
 Unchecked Deallocation
 ----------------------
 
-.. admonition:: Relevant topics
+So far, we've seen multiple examples of using :ada:`new` to allocate objects.
+In this section, we discuss how to manually deallocate objects.
 
-    - `Unchecked Storage Deallocation <http://www.ada-auth.org/standards/2xrm/html/RM-13-11-2.html>`_
+Our starting point to manually deallocate an object is the generic
+:ada:`Unchecked_Deallocation` procedure. We first instantiate this procedure
+for an access type whose objects we want to be able to deallocate. For example,
+let's instantiate it for the :ada:`Integer_Access` type:
 
-.. todo::
+.. code:: ada compile_button project=Courses.Advanced_Ada.Access_Types.Simple_Unchecked_Deallocation
+
+    with Ada.Unchecked_Deallocation;
+
+    package Integer_Types is
+
+       type Integer_Access is access Integer;
+
+       --
+       --  Instantiation of Unchecked_Deallocation
+       --  for the Integer_Access type:
+       --
+       procedure Free is
+         new Ada.Unchecked_Deallocation
+           (Object => Integer,
+            Name   => Integer_Access);
+    end Integer_Types;
+
+Here, we declare the :ada:`Free` procedure, which we can then use to deallocate
+objects that were allocated for the :ada:`Integer_Access` type.
+
+:ada:`Unchecked_Deallocation` is a generic procedure that we can instantiate
```

Comment:
I suggest :ada:`Ada.Unchecked_Deallocation` throughout, to avoid confusion with the old name that is still legal and available.